pkg/build/rpm_test.go

```diff
@@ -12,113 +13,252 @@ import (
-	err = builder.copyRPMs()
+	err = builder.writeRPMScript(RPMFileNames)
 
 	// Verify
 	require.NoError(t, err)
 
-	_, err = os.Stat(filepath.Join(rpmDestDir, "rpm1.rpm"))
+	expectedFilename := filepath.Join(builder.combustionDir, modifyRPMScriptName)
+	foundBytes, err := os.ReadFile(expectedFilename)
 	require.NoError(t, err)
 
-	_, err = os.Stat(filepath.Join(rpmDestDir, "rpm2.rpm"))
+	stats, err := os.Stat(expectedFilename)
 	require.NoError(t, err)
+	assert.Equal(t, fs.FileMode(modifyScriptMode), stats.Mode())
 
-	// Cleanup
-	assert.NoError(t, file1.Close())
-	assert.NoError(t, file2.Close())
+	foundContents := string(foundBytes)
+	assert.Contains(t, foundContents, "rpm1.rpm")
+	assert.Contains(t, foundContents, "rpm2.rpm")
 }
 
-func TestGetRPMFileNamesNoRPMs(t *testing.T) {
+func TestProcessRPMs(t *testing.T) {
 	// Setup
+	tmpDir, err := os.MkdirTemp("", "eib-process-RPMs-test-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	RPMSourceDir := filepath.Join(tmpDir, "rpms")
+	err = os.Mkdir(RPMSourceDir, 0o755)
+	require.NoError(t, err)
+
 	bc := config.BuildConfig{
-		ImageConfigDir: "../config/testdata",
+		ImageConfigDir: tmpDir,
 	}
 	builder := New(nil, &bc)
-	err := builder.prepareBuildDir()
+	err = builder.prepareBuildDir()
 	require.NoError(t, err)
-	defer os.Remove(builder.eibBuildDir)
 
-	rpmSourceDir := filepath.Join(builder.buildConfig.ImageConfigDir, "rpms")
+	file1Path := filepath.Join(RPMSourceDir, "rpm1.rpm")
+	file1, err := os.Create(file1Path)
+	require.NoError(t, err)
+	defer file1.Close()
+
+	file2Path := filepath.Join(RPMSourceDir, "rpm2.rpm")
+	file2, err := os.Create(file2Path)
+	require.NoError(t, err)
+	defer file2.Close()
 
 	// Test
-	rpmFileNames, err := builder.getRPMFileNames(rpmSourceDir)
+	err = builder.processRPMs()
 
 	// Verify
-	require.ErrorContains(t, err, "no rpms found")
+	require.NoError(t, err)
+
+	_, err = os.Stat(filepath.Join(builder.combustionDir, "rpm1.rpm"))
+	require.NoError(t, err)
+
+	_, err = os.Stat(filepath.Join(builder.combustionDir, "rpm2.rpm"))
+	require.NoError(t, err)
+
+	expectedFilename := filepath.Join(builder.combustionDir, modifyRPMScriptName)
+	foundBytes, err := os.ReadFile(expectedFilename)
+	require.NoError(t, err)
 
-	assert.Empty(t, rpmFileNames)
+	foundContents := string(foundBytes)
+	assert.Contains(t, foundContents, "rpm1.rpm")
+	assert.Contains(t, foundContents, "rpm2.rpm")
 }
 
-func TestCopyRPMsNoRPMDir(t *testing.T) {
+func TestGenerateRPMPath(t *testing.T) {
 	// Setup
+	tmpDir, err := os.MkdirTemp("", "eib-generate-RPM-path-test-")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	RPMSourceDir := filepath.Join(tmpDir, "rpms")
+	err = os.Mkdir(RPMSourceDir, 0o755)
+	require.NoError(t, err)
+
 	bc := config.BuildConfig{
-		ImageConfigDir: "../config/ThisDirDoesNotExist",
+		ImageConfigDir: tmpDir,
 	}
 	builder := New(nil, &bc)
-	err := builder.prepareBuildDir()
+
+	expectedPath := filepath.Join(builder.buildConfig.ImageConfigDir, "rpms")
```

Comment:
The path that we expect is `RPMSourceDir` - not the one that the builder internally builds. We should not confuse test expectations with how the builder is initialized.